### PR TITLE
Knntest

### PR DESCRIPTION
**Pull request by siebenHeaven**
https://github.com/shogun-toolbox/shogun/issues/2024
Unit Test regarding this issue.

### CODE REVIEW

File: tests/unit/multiclass/KNN_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	labels->set_label(2, 1);
+	labels->set_label(3, 0);
+	CDenseFeatures<float64_t>* features=new CDenseFeatures<float64_t>(rect);
+	SG_REF(features);
+	CEuclideanDistance* distance=new CEuclideanDistance(features, features);
+	CKNN* knn=new CKNN(4, distance, labels);
+    
+	knn->train(features);
+    
+    /*Test data*/
```
Comment (karlnapf) [inline tests/unit/multiclass/KNN_unittest.cc]: Indentation

File: tests/unit/multiclass/KNN_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	CDenseFeatures<float64_t>* test=new CDenseFeatures<float64_t>(data);
+
+	CMulticlassLabels* result=knn->apply_multiclass(test);
+
+	EXPECT_EQ(1.000000, result->get_label(0));
+	
+	SG_UNREF(knn);
+	SG_UNREF(features);
+
+	}
```
Comment (karlnapf) [inline tests/unit/multiclass/KNN_unittest.cc]: Indentation

File: tests/unit/multiclass/KNN_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	CDenseFeatures<float64_t>* features=new CDenseFeatures<float64_t>(rect);
+	SG_REF(features);
+	CEuclideanDistance* distance=new CEuclideanDistance(features, features);
+	CKNN* knn=new CKNN(4, distance, labels);
+    
+	knn->train(features);
+    
+    /*Test data*/
+	SGMatrix<float64_t> data(2, 1);
+	data(0,0)=0;
```
Comment (karlnapf) [inline tests/unit/multiclass/KNN_unittest.cc]: What about a few more points?
Also, can you comment on the rationale behind this test?

File: tests/unit/multiclass/KNN_unittest.cc [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/multiclass/KNN.h>
+#include <shogun/distance/EuclideanDistance.h>
+#include <gtest/gtest.h>
+#include <shogun/labels/BinaryLabels.h>
+
+using namespace shogun;
+
+TEST (KNN, simple_classification)
```
Comment (karlnapf) [inline tests/unit/multiclass/KNN_unittest.cc]: The test name is not really useful, make it more explixit

**Comment by karlnapf** [comment]
Great thanks for that!
This test is very high level, can we test some of the subcodes of knn?

**Comment by siebenHeaven** [comment]
Yeah, I'll try writing a few more tests for different subcodes.

File: tests/unit/multiclass/KNN_unittest.cc (hunk 4 above)
Comment (siebenHeaven) [inline tests/unit/multiclass/KNN_unittest.cc]: Could you suggest a better name? I'll get an idea of how tests should be named then.

**Comment by karlnapf** [comment]
Updates? :)
Would be good to get that merged as one of our GSoC students is currently working on speeding up KNN

**Pull request closed without merging**